pkg/buildpack/buildpack_test.go

```diff
@@ -548,6 +563,172 @@ version = "1.2.3"
+		)
+
+		it.Before(func() {
+			tmpDir, err = os.MkdirTemp("", "")
+			h.AssertNil(t, err)
+		})
+
+		it.After(func() {
+			err := os.RemoveAll(tmpDir)
+			if runtime.GOOS != "windows" {
+				// avoid "The process cannot access the file because it is being used by another process"
+				// error on Windows
+				h.AssertNil(t, err)
+			}
+		})
+
+		when("BuildModule contains N flattened buildpack", func() {
+			it("returns N tar files", func() {
+				bp := buildpack.FromBlob(
+					&dist.BuildpackDescriptor{
+						WithAPI: api.MustParse("0.3"),
+						WithInfo: dist.ModuleInfo{
+							ID:      "buildpack-1-id",
+							Version: "buildpack-1-version-1",
+							Name:    "buildpack-1",
+						},
+					},
+					&readerBlob{
+						openFn: func() io.ReadCloser {
+							tarBuilder := archive.TarBuilder{}
+
+							// Buildpack 1
+							tarBuilder.AddDir("/cnb/buildpacks/buildpack-1-id", 0700, time.Now())
+							tarBuilder.AddDir("/cnb/buildpacks/buildpack-1-id/buildpack-1-version-1", 0700, time.Now())
+							tarBuilder.AddFile("/cnb/buildpacks/buildpack-1-id/buildpack-1-version-1/buildpack.toml", 0700, time.Now(), []byte(`
+api = "0.3"
+
+[buildpack]
+id = "buildpack-1-id"
+version = "buildpack-1-version-1"
+
+`))
+							tarBuilder.AddDir("/cnb/buildpacks/buildpack-1-id/buildpack-1-version-1/bin", 0700, time.Now())
+							tarBuilder.AddFile("/cnb/buildpacks/buildpack-1-id/buildpack-1-version-1/bin/detect", 0700, time.Now(), []byte("detect-contents"))
+							tarBuilder.AddFile("/cnb/buildpacks/buildpack-1-id/buildpack-1-version-1/bin/build", 0700, time.Now(), []byte("build-contents"))
+
+							// Buildpack 2
+							tarBuilder.AddDir("/cnb/buildpacks/buildpack-2-id", 0700, time.Now())
+							tarBuilder.AddDir("/cnb/buildpacks/buildpack-2-id/buildpack-2-version-1", 0700, time.Now())
+							tarBuilder.AddFile("/cnb/buildpacks/buildpack-2-id/buildpack-2-version-1/buildpack.toml", 0700, time.Now(), []byte(`
+api = "0.3"
+
+[buildpack]
+id = "buildpack-2-id"
+version = "buildpack-2-version-1"
+
+`))
+							tarBuilder.AddDir("/cnb/buildpacks/buildpack-2-id/buildpack-2-version-1/bin", 0700, time.Now())
+							tarBuilder.AddFile("/cnb/buildpacks/buildpack-2-id/buildpack-2-version-1/bin/detect", 0700, time.Now(), []byte("detect-contents"))
+							tarBuilder.AddFile("/cnb/buildpacks/buildpack-2-id/buildpack-2-version-1/bin/build", 0700, time.Now(), []byte("build-contents"))
+
+							return tarBuilder.Reader(archive.DefaultTarWriterFactory())
+						},
+					},
+					buildpack.Flattened(),
+				)
+
+				tarPaths, err := buildpack.ToNLayerTar(tmpDir, bp, logger)
+				h.AssertNil(t, err)
+				h.AssertEq(t, len(tarPaths), 2)
+				for _, tarPath := range tarPaths {
+					h.AssertOnTarEntry(t, tarPath.Path(), fmt.Sprintf("/cnb/buildpacks/%s", tarPath.Info().ID),
+						h.IsDirectory(),
+					)
+					h.AssertOnTarEntry(t, tarPath.Path(), fmt.Sprintf("/cnb/buildpacks/%s/%s", tarPath.Info().ID, tarPath.Info().Version),
+						h.IsDirectory(),
+					)
```

Comment:
Something about this test is a little off... I was going to suggest making the expectation a bit more robust by calculating the expected "buildpack-XXX-id" and "buildpack-XXX-version-1" from the loop index, but it didn't work. Then when I added `fmt.Println("tarPath.Info().ID:", tarPath.Info().ID)` to the beginning of the loop I saw 
```
tarPath.Info().ID: buildpack-2-id
tarPath.Info().ID: buildpack-2-id
```
The same ID prints out twice...